### ISSUE TITLE
Stutter in regular intervals?

### ISSUE THREAD

**Issue by 2mg** [bolrog/d2dx#118]
I'm getting a smooth gameplay, but with stutters in regular intervals exactly every few seconds (see spikes in frametime):

![stutterd2dx](https://user-images.githubusercontent.com/7749968/123570590-cf859000-d7c8-11eb-9cb9-74ab73891076.jpg)

![stutterd2dxnovsync](https://user-images.githubusercontent.com/7749968/123570593-d14f5380-d7c8-11eb-9c6e-2f8bef8a5c90.jpg)

1.13x (both c & d, haven't tried 1.14x).
Win7x64 with or without DWM.exe (Aero composition).
With or without FPS limiter.

**Comment by eskiq** [comment]
Game for me is also very smooth, but every 4-5 seconds there is stutter. Never seen d2 to be so smooth :D. After walking for some time in lets say act 2 city, those spikes are gone for me in the area where I am walking. It's not a solution at all as those spikes will appear constantly until I do some walking in circles. Maybe after some time it's correctly buffered or something.

**Comment by 2mg** [comment]
> 
> 
> Game for me is also very smooth, but every 4-5 seconds there is stutter. Never seen d2 to be so smooth :D. After walking for some time in lets say act 2 city, those spikes are gone for me in the area where I am walking. It's not a solution at all as those spikes will appear constantly until I do some walking in circles. Maybe after some time it's correctly buffered or something.

Yeah, walking around, especially out of towns, seems to calm it a bit, but the graph always show they are present, just shorter.

**Comment by 2mg** [comment]
I can confirm that this doesn't happen on Win10 though. Same PC, same drivers, same game and settings.

**Comment by bolrog** [comment]
Strange.. I will try to think about what this could be.

**Comment by 2mg** [comment]
> 
> 
> Strange.. I will try to think about what this could be.

Just note that W10 doesn't have it, while W7 does, REGARDLESS of Vsync setting on both systems. I'd say it has to do with W7 DWM and W10 DXGI FLIP, but I can't be sure, as this issue _isn't affected with Vsync_, so it might be something with Win7 only. I mean, the more you run around, or just leave the game running, the more the chance it will calm down - on W7, which is really odd.

Thanks for looking into it!

**Comment by Aaron03071991** [comment]
Ill drop my few cents to the topic as well.
Laptop with i7-7700hq, hd630 (tho this laptop also have nvidia gtx1050, no difference on which gpu D2 runs), w10.
Game overall runs flawlessly, almost no spikes, unless game displays text on screen. What do i mean is, lets say i move from river of flame to chaos sanctuary, and in that very moment when i change map, game displays message about it, then i get huge 1 sec hang, then all goes back to normal. I can produce such lag spikes all day, enough ill stand on the transition ground between 2 "maps" and move from one to another. If i didnt explain it well enough, i can record video of this.
I tried tinkering with D2 settings, D2DX settings, gpu settings, no change, game with plugy or without, always the same result.
Ah! This happens in 1.13d and 1.14d, not sure about other versions. Windows up to date, with all possible updates, drivers also up to date.

**Comment by snifflezak** [comment]
I have the stutter at consistent intervals in Win10, but my hardware is fairly old. 
Phenom 1055t, m4a89gtd pro/usb3, 750ti, Win10 Pro
1.14d LOD

edit:
so i dualbooted to my second windows 10 os on this same pc. it's the same build and everything but helps for testing.

for whatever reason widescreen does not work on windows # 2.
BUT the stutter isn't there, so I don't know if that helps lol but I guess it's not a hardware issue.
